policy/modules/system/systemd.te

```diff
@@ -489,6 +520,125 @@ optional_policy(`
 	miscfiles_read_localization(systemd_generator_t)
 ')
 
+#######################################
+#
+# systemd-homed policy
+#
+
+dontaudit systemd_homed_t self:capability { sys_resource sys_admin };
+allow systemd_homed_t self:netlink_kobject_uevent_socket create_socket_perms;
+
+nnp_domtrans_pattern(systemd_homed_t, systemd_homework_exec_t, systemd_homework_t)
+
+allow systemd_homed_t systemd_homed_tmpfs_t:file manage_file_perms;
+fs_tmpfs_filetrans(systemd_homed_t, systemd_homed_tmpfs_t, file)
+
+manage_sock_files_pattern(systemd_homed_t, systemd_userdbd_runtime_t, systemd_homed_runtime_t)
+manage_dirs_pattern(systemd_homed_t, systemd_homed_runtime_t, systemd_homed_runtime_t)
+filetrans_pattern(systemd_homed_t, systemd_userdbd_runtime_t, systemd_homed_runtime_t, sock_file)
```

Comment:
This is highly subjective but I thought I at least mention it briefly.

I am not sure if it is worth the trouble to label the content in `/run/systemd/userdb` with private types. Mainly because most of these sockets are used for name and group resolving (whether the name is dynamic, a containerr name, or a name of a user managed by systemd-homed) eventually it boils down to either nss-user/group equivalent resolving or nss-hosts equivalent resolving.

The userdbd socket (or atleast one of them) is a notable exception to this rule as it is used to provide the functionality that systemd-userdbctl provides either directly or indirectly AFAIK.

Regardless though the socket end points are always different and so you  could just allow writes to all sockets in `/run/systemd/userdb` for simplicity and then still use the `connectto` pattern to narrow down how the socket writes can effectively be used.

Its a matter of taste, this is not wrong in any way but it is expensive

Reply:
I'll keep it as-is. It's a small size and we can revise to make it only the sockets that filetrans in the future, if necessary.